dashboard/app/controllers/programming_classes_controller.rb

```diff
@@ -58,6 +59,15 @@ def show
     @programming_environment_categories = @programming_class.programming_environment.categories_for_navigation
   end
 
+  def show_by_keys
+    return render :not_found unless @programming_class
+    if params[:programming_environment_name] && params[:programming_class_key]
```

Comment:
Does this check guarantee that `@programming_class` is non-nil?  Would it be more appropriate to check `if @programming_class` here instead?

Reply:
Yeah, I think this is redundant with the lines above plus `set_class_by_keys`. I'll remove!